plugins/destination/postgresql/client/insert.go

```diff
@@ -34,6 +35,33 @@ func (c *Client) pgTables(ctx context.Context) (map[string]struct{}, error) {
 	return list, nil
 }
 
+type tableDetails map[string]tableDetail
+type tableDetail struct {
+	indexes []int
+	values  map[string]bool
+	sql     string
+}
+
+// When a batch is sent and inserted into the DB we need to reset the PKValues
+func (p *tableDetails) resetValues() {
+	for _, v := range *p {
+		v.values = make(map[string]bool)
+	}
+}
+
+// extracts PKValues from an Arrow record into a string
+// An arrow record may have multiple rows and multiple PK columns so a list of pk value strings is returned
+func getPKValues(r arrow.Record, pkIndex []int) []string {
+	results := make([]string, r.NumRows())
+	for ri := range results {
+		for _, i := range pkIndex {
+			col := r.Column(i)
+			results[ri] += col.String()
```

Comment:
Edge case, but it seems like we need to use a 2D `[][]string` here, otherwise these two rows will be counted as a collision:


row | pk_col1 | pk_col2 |
----|-----|-----|
1     | aa    | b
2    | a      | ab

In this case it wouldn't really matter since it will just cause a premature flush, but there is also a warning message, so it would be better if it was correct in all cases

Reply:
Nice catch. I fixed this by adding the col index to the stringified value. This way the pk value can continue to be the map key